### PR TITLE
nixops: remove (v1.x)

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

NOTE: a preferable solution may be feasible #154715

The NixOps 1.x branch has been unsupported for a long time, breaks
very often in Nixpkgs, because it requires heavily outdated deps,
and has put undue burden on the python packages maintainers.

The 1.x branch is unsustainable and it is time that we acknowledge
this fact. The only way forward for NixOps is the unreleased
version 2.0, which needs testing and fixes before it can be
released and become the replacement for the default nixops package.

As mentioned before: https://github.com/NixOS/nixpkgs/issues/144627#issuecomment-975900718

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @aminechikhaoui @eelco @rob

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -32548,7 +32548,7 @@ with pkgs;
 
   nixStatic = pkgsStatic.nix;
 
-  nixops = callPackage ../tools/package-management/nixops { };
+  nixops = builtins.throw "The NixOps 1.x branch is unmaintained and unsupported. Please upgrade to nixopsUnstable and contribute to the NixOps 2 release.";
```
Comment (Mindavi) [inline pkgs/top-level/all-packages.nix:32551]: This can go in aliases, I think.

Comment (roberth) [reply pkgs/top-level/all-packages.nix:32551]: I don't see why that would be useful. Isn't it nicer to keep the nixops attrs together? Will make the 2.0 release slightly easier.

Comment (jonringer) [reply pkgs/top-level/all-packages.nix:32551]: We use aliases to prevent all-packages.nix from being cluttered with 100's of deprecation/removal errors.

**Comment by siraben** [comment]
Any updates on this?

**Comment by adisbladis** [comment]
Fwiw: I think this is sad and a testament to how badly Python package maintenance works in nixpkgs.
We have all the technical requisites for these problems _never_ to occur, yet here we are.

**Comment by Artturin** [comment]
Solved by https://github.com/NixOS/nixpkgs/pull/154715

Thanks all!

**Pull request closed without merging**